File: frontend/caixa-installments/src/app/models/installments.ts
export class Installment {
    id?: string;
    nroContrato?: string;
    dueDate?: Date;
    paid?: boolean;
    amount?: number;
    interest?: number;
    insurance?: number;
    fees?: number;
    outstandingBalance?: number;
    monetaryCorrection?: number;
    amortization?: number;
    isContribution?: boolean;

    constructor(data: any) {
        this.convertAndAssign(data);
    }

    private convertAndAssign(data: any): void {
        this.id = data.id,
        this.nroContrato = data.nroContrato;
        this.dueDate = new Date(data.dueDate);
        this.paid = data.paid === 'true';  
        this.amount = parseFloat(data.amount);
        this.interest = parseFloat(data.interest);
        this.insurance = parseFloat(data.insurance);
        this.fees = parseFloat(data.fees);
        this.outstandingBalance = parseFloat(data.outstandingBalance);
        this.monetaryCorrection = parseFloat(data.monetaryCorrection);
        this.amortization = parseFloat(data.amortization);
        this.isContribution = data.isContribution === 'true'; 
    }
}

File: backend/src/main/kotlin/br/com/ricas/caixa/domain/entity/Installments.kt
package br.com.ricas.caixa.domain.entity

import br.com.ricas.caixa.application.api.request.InstallmentRequest
import br.com.ricas.caixa.application.api.response.InstallmentResponse
import br.com.ricas.caixa.infrastructure.entity.InstallmentsEntity
import java.util.Date

data class Installments(
    val nroContrato: String?,
    val dueDate: Date?,
    val paid: Boolean?,
    val amount: Double?,
    val interest: Double?,
    val insurance: Double?,
    val fees: Int?,
    val outstandingBalance: Double?,
    val monetaryCorrection: Double?,
    val amortization: Double?,
    val isContribution: Boolean?
) {
    companion object {
        fun toDomain(installmentRequest: InstallmentRequest): Installments {
            return Installments(
                nroContrato = installmentRequest.nroContrato,
                dueDate = installmentRequest.dueDate,
                paid = installmentRequest.paid,
                amount = installmentRequest.amount,
                interest = installmentRequest.interest,
                insurance = installmentRequest.insurance,
                fees = installmentRequest.fees,
                outstandingBalance = installmentRequest.outstandingBalance,
                monetaryCorrection = installmentRequest.monetaryCorrection,
                amortization = installmentRequest.amortization,
                isContribution = installmentRequest.isContribution
            )
        }

    }
    fun toResponse(): InstallmentResponse {
        return InstallmentResponse(
            nroContrato = nroContrato,
            dueDate = dueDate,
            paid = paid,
            amount = amount,
            interest = interest,
            insurance = insurance,
            fees = fees,
            outstandingBalance = outstandingBalance,
            monetaryCorrection = monetaryCorrection,
            amortization = amortization,
            isContribution = isContribution
        )
    }


    fun toEntity(): InstallmentsEntity {
        return InstallmentsEntity(
            id = null,
            nroContrato = nroContrato,
            dueDate = dueDate,
            paid = paid,
            amount = amount,
            interest = interest,
            insurance = insurance,
            fees = fees,
            outstandingBalance = outstandingBalance,
            monetaryCorrection = monetaryCorrection,
            amortization = amortization,
            isContribution = isContribution
        )
    }
}
